### DISCUSSION TITLE
eslint/json´s documentation and typescript-eslint

### DISCUSSION THREAD

**Discussion by Byron2016** [eslint/json#33] [Q&A]
I was doing some tests with **@eslint/json**  and **typescript-eslint** and I had some problems, then when I went to documentation, I saw there are not examples about how to use it in a typescript project and also some diferencies with examples that confuse me then I decided to let you know this disccusion, I hope it would be interesting your you and let you see that with some small changes it could improve for people like me.

# 1.- Documentation
Inside **@eslint/json** documentation, there are two examples:
**example 1**
```js
import json from "@eslint/json";
export default [
	{
		plugins: {
			json, //<-----
		},
	},

	// lint JSON files
	{
		files: ["**/*.json"],
		language: "json/json",
		rules: {
			"json/no-duplicate-keys": "error",
		},
	},
	// ......
];
```
**example 2: Recommended Configuration**
```js
import json from "@eslint/json";

export default [
	// lint JSON files
	{
		files: ["**/*.json"],
		ignores: ["package-lock.json"],
		language: "json/json",
		...json.configs.recommended,
	},

	// ....
];
```
- What is the diference between to have `{plugins: {json,},},` section (example 1) or not (example 2)?
- What do this **plugins seccions**  `{plugins: {json,},},` (example 1))?
- Could be a good idea to add a comment to explain pleople about this  **plugins secction**?

# 2.- Use of  **@eslint/json**  and **typescript-eslint**
I don´t know if other people have problems understanding the best way to use it, it is very common with manuals that they are written by experienced people with a big knowledge of the tool or experience working with it, but this manuals are not reviewed for people with less experience, then a manual that is perfect for first user is not enough for second kind of people.
**typescript-eslint** has a utility named **config** (A utility function for creating type-safe flat configs) then my propuse is : 
- to add a example inside **@eslint/json** documentation  of how to use **@eslint/json** with **typescript-eslint** 
  - using this utility function 
  - without this utility function. 
 
Inside **typescript-eslint** documentation [...config()](https://typescript-eslint.io/packages/typescript-eslint/#config) there are a note "...**strongly** recommend using this utility to improve the config authoring experience...", they also add a section to use without this utily function but it add some jdocs annotations that are confusing.

# 3.- Translations
Eslint is a tool very popular inside developers, maybe it is the most used utility for linting around the World, then what do you think about to add a directory with the posibility to add README files with translations to other languages? This could be doing for community colaborations.
I have created a [PR-35](https://github.com/eslint/json/pull/35)

I will be waiting your response with expectation.

**Comment by nzakas** [comment]
> * What is the diference between to have `{plugins: {json,},},` section (example 1) or not (example 2)?

`json.configs.recommended` already includes the `plugins`, so you don't need to manually include it.

> * What do this **plugins seccions**  `{plugins: {json,},},` (example 1))?
> 
> * Could be a good idea to add a comment to explain pleople about this  **plugins secction**?

This is just the way plugins work in ESLint. See the docs: https://eslint.org/docs/latest/use/configure/plugins

For typescript-eslint, you'll have to talk to the maintainers about how to correctly use it. We aren't involved with that project.
